### DISCUSSION TITLE
Document "magic" of how Dagger SDKs ensure that a Dagger Engine exists without an explicit install

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#12501] [Ideas]
### What is the issue?

https://discord.com/channels/707636530424053791/1110935342577631253/1113877665192087643

From @ringods
> you might want to highlight that the Dagger SDK will do some magic behind the scenes to ensure you have a Dagger Engine running in each of the CI integrations here: https://docs.dagger.io/7442989/cookbook#integrations
I was puzzled that I didn't need to install dagger as part of e.g. my Github Actions workflow.

**Comment by helderco** [comment]
That feature is called "Automatic Provisioning". There's more info about it in the [SDK contribution guide](https://github.com/dagger/dagger/blob/main/sdk/CONTRIBUTING.md#architecture).
